### PR TITLE
CytoSnake Bug Fixes From Cell-Health Data Usage

### PR DESCRIPTION
**Pull request by axiomcura**
_modified from [EmbeddedArtistry](https://embeddedartistry.com/blog/2017/08/04/a-github-pull-request-template-for-your-projects/)_

# Description

Thank you for your contribution to CytoSnake!
Please _succinctly_ summarize your proposed change.
What motivated you to make this change?

Please also link to any relevant issues that your code is associated with.

This PR fixes bugs that were found and squashed on the [cell-health dataset ](https://github.com/WayScience/cell-health-data) when using `CytoSnake` locally. 

The next step is to make a release after this PR is merged, and use it as a dependency for the [cell-health dataset ](https://github.com/WayScience/cell-health-data)

## What is the nature of your change?

- [X] Bug fix (fixes an issue).
- [ ] Enhancement (adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] This change requires a documentation update.

# Checklist

Please ensure that all boxes are checked before indicating that a pull request is ready for review.

- [X] I have read the [CONTRIBUTING.md](CONTRIBUTING.md) guidelines.
- [X] My code follows the style guidelines of this project.
- [X] I have performed a self-review of my own code.
- [X] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [X] I have deleted all non-relevant text in this pull request template.

### CODE REVIEW

File: configs/wf_configs/cp_process.yaml [hunk 1]
```diff
@@ -10,6 +10,10 @@ docs: |
   ---------------
   Below the workflow steps are separated in chunks.
 
+  single_cell_config:
+    Parameters used for SingleCell object, used for interacting with single cell data.
+    For more documenation about SingelCell please refer to here:
```
Comment (gwaybio) [inline configs/wf_configs/cp_process.yaml:15]:
```suggestion
    For more documentation about SingleCell please refer here:
```

File: workflows/rules/aggregate.smk [hunk 2]
```diff
@@ -25,13 +25,14 @@ rule aggregate:
         barcodes=BARCODES,
         metadata=METADATA_DIR,
     output:
-        aggregate_profile=get_data_path(input_type="aggregated", tolist=True),
-        cell_counts=CELL_COUNTS,
+        aggregate_profile=get_data_path(input_type="aggregated"),
+        cell_counts=get_data_path(input_type="cell_counts"),
     log:
-        "logs/aggregate_{file_name}.log",
+        "logs/aggregate_{basename}.log",
```
Comment (gwaybio) [inline workflows/rules/aggregate.smk:31]: just confirming `basename` is defined elsewhere

File: workflows/rules/aggregate.smk [hunk 3]
```diff
@@ -25,13 +25,14 @@ rule aggregate:
         barcodes=BARCODES,
         metadata=METADATA_DIR,
     output:
-        aggregate_profile=get_data_path(input_type="aggregated", tolist=True),
-        cell_counts=CELL_COUNTS,
+        aggregate_profile=get_data_path(input_type="aggregated"),
+        cell_counts=get_data_path(input_type="cell_counts"),
```
Comment (gwaybio) [inline workflows/rules/aggregate.smk:29]: confirming `get_data_path()` is defined somewhere

Comment (gwaybio) [reply workflows/rules/aggregate.smk:29]: ah, this is defined in `common.smk`, gotcha

File: workflows/scripts/aggregate_cells.py [hunk 4]
```diff
@@ -79,12 +64,15 @@ def aggregate(
...
 
-    # generating cell count
+    # loading in barcodes
     logging.info(f"Loading barcodes from: {barcode_path}")
     barcode_platemap_df = pd.read_csv(barcode_path)
 
+    # plate map search
     logging.info("Selecting associated plate map")
     try:
+        # plate varaible is used in the query string below
```
Comment (gwaybio) [inline workflows/scripts/aggregate_cells.py:74]:
```suggestion
        # plate variable is used in the query string below
```

File: workflows/scripts/aggregate_cells.py [hunk 5]
```diff
@@ -94,28 +82,30 @@ def aggregate(
             "Unable to find associated platemap name with given plate barcode"
         )
 
+    # loading in indentified platemap
```
Comment (gwaybio) [inline workflows/scripts/aggregate_cells.py:85]: Not sure what you mean by `indentified` (identified?)

File: workflows/scripts/aggregate_cells.py [hunk 6]
```diff
@@ -146,16 +140,27 @@ def aggregate(
...
     log_path = str(snakemake.log)
 
+    # print(plate_data)
+    # print(barcode_path)
+    # print(metadata_dir_path)
+    # print(cell_count_output)
+    # print(aggregate_output)
+    # print(configs)
+    # print("#################3")
+    # print(log_path)
```
Comment (gwaybio) [inline workflows/scripts/aggregate_cells.py:154]: probably to delete

**Review by gwaybio — APPROVED** [review]
Minor comments, mostly typo fixes

File: workflows/scripts/aggregate_cells.py (hunk 6 above)
Comment (axiomcura) [reply workflows/scripts/aggregate_cells.py:154]: I totally forgot to remove this. This is an example of my furstration.

File: workflows/rules/aggregate.smk (hunk 2 above)
Comment (axiomcura) [reply workflows/rules/aggregate.smk:31]: yes, it is defined in the common.smk file

File: workflows/scripts/aggregate_cells.py (hunk 5 above)
Comment (axiomcura) [reply workflows/scripts/aggregate_cells.py:85]: Yes it was a type! fixed!

**Comment by axiomcura** [comment]
Alright all comments have been addressed! Merging !

**Pull request merged**